### DISCUSSION TITLE
How does Java connect to remote JanusGraph and can obtain JanusGraphManagement to operate transactions

### DISCUSSION THREAD

**Discussion by qDonl** [JanusGraph/janusgraph#2647] [Q&A]
## Desc
I am going to use Java to connect to the remote JanusGraph, using [the official tutorial](https://docs.janusgraph.org/connecting/java/), where the hosts in `conf/remote-objects.yaml` have been changed to my remote server address.
But after the connection is complete, I see that the graph address obtained through `Graph graph = g.getGraph()` is `127.0.0.1`

I also tried to use the configuration to connect to the remote database
```java
JanusGraphFactory.Builder config = JanusGraphFactory.build();
config.set("storage.hostname", VM_62);
config.set("storage.backend", "hbase");
config.set("index.search.backend", "elasticsearch");
config.set("index.search.hostname", VM_62);
config.set("index.search.port", "9200");
config.set("graph.graphname", "graph2");

StandardJanusGraph graph = (StandardJanusGraph) config.open();
```
However, the remote JanusGraph still cannot be connected in this way (only the properties of `.properties` can be configured)

## My Questions
1. How to connect to remote JanusGraph
2. How to get JanusGraphManagement to operate transactions after successful connection?

Thx.

**Comment by vtslab** [comment]
1. The EmptyGraph instance resides on your local machine. Also, the GraphTraversalSource g resides on your local machine and getGraph() returns the local graph. However, operations on a GraphTraversalSource created using withRemote() are executed on the remote graph.
2. By default every remote request is wrapped in its own transaction on the server. Therefore, during bulk loading it is advised to combine multiple addV and addE steps in a single remote request (g.addV().addV().addV() etc.) It is possible to sidestep the default behaviour, see: [https://tinkerpop.apache.org/docs/current/reference/#sessions](https://tinkerpop.apache.org/docs/current/reference/#sessions).
